central/debug/service/service.go

```diff
@@ -343,6 +373,90 @@ func (s *serviceImpl) getLogImbue(zipWriter *zip.Writer) error {
 	return err
 }
 
+func (s *serviceImpl) getAuthProviders(_ context.Context) (interface{}, error) {
+	authProviders := s.authProviderRegistry.GetProviders(nil, nil)
+
+	var storageAuthProviders []*storage.AuthProvider
+	for _, authProvider := range authProviders {
+		storageAuthProviders = append(storageAuthProviders, authProvider.StorageView())
+	}
+
+	return storageAuthProviders, nil
+}
+
+func (s *serviceImpl) getGroups(_ context.Context) (interface{}, error) {
+	accessGroupsCtx := sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Access)))
+
+	return s.groupDataStore.GetAll(accessGroupsCtx)
+}
+
+type diagResolvedRole struct {
+	Role          *storage.Role              `json:"role,omitempty"`
+	PermissionSet map[string]string          `json:"permission_set,omitempty"`
+	AccessScope   *storage.SimpleAccessScope `json:"access_scope,omitempty"`
+}
+
+func (s *serviceImpl) getRoles(_ context.Context) (interface{}, error) {
+	accessRolesCtx := sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Access)))
+
+	roles, errGetRoles := s.roleDataStore.GetAllRoles(accessRolesCtx)
+	if errGetRoles != nil {
+		return nil, errGetRoles
+	}
+
+	var resolvedRoles []*diagResolvedRole
+	for _, role := range roles {
+		diagRole := diagResolvedRole{
+			Role: role,
+		}
+
+		if resolvedRole, err := s.roleDataStore.GetAndResolveRole(accessRolesCtx, role.Name); err == nil {
```

Comment:
@mtodor @rukletsov 
For some reason I get panic on this code. It looks like `GetAndResolveRole` can retrun `nil, nil` when read is not allowed 

https://github.com/stackrox/stackrox/blob/5a5060fc4a1bcb0de1981614ad87be72ee77f541/central/role/datastore/datastore_impl.go#L375-L379

Reply:
```
2022/05/19 11:08:35 http: panic serving 10.100.98.212:14591: runtime error: invalid memory address or nil pointer dereference
goroutine 97521 [running]:
net/http.(*conn).serve.func1()
	net/http/server.go:1801 +0xb9
panic({0x5a073e0, 0x95ce170})
	runtime/panic.go:1047 +0x266
github.com/stackrox/rox/pkg/grpc/metrics.(*httpMetricsImpl).WrapHandler.func1.1()
	github.com/stackrox/rox/pkg/grpc/metrics/http_metrics_impl.go:61 +0x151
panic({0x5a073e0, 0x95ce170})
	runtime/panic.go:1038 +0x215
github.com/stackrox/rox/central/debug/service.(*serviceImpl).getRoles(0xc000704460, {0x60aedf7, 0x2})
	github.com/stackrox/rox/central/debug/service/service.go:422 +0x2dd
github.com/stackrox/rox/central/debug/service.fetchAndAddJSONToZip({0x6c2c270, 0xc0075fbfb0}, 0xc0075fbfb0, {0x612aac7, 0x19}, 0x0)
	github.com/stackrox/rox/central/debug/service/service.go:238 +0x46
github.com/stackrox/rox/central/debug/service.(*serviceImpl).writeZippedDebugDump(0xc000704460, {0x6c2c270, 0xc0075fbfb0}, {0x6be04f0, 0xc0069e7f20}, {0xc006f7d920, 0x2b}, {0x3, 0x2, 0x0, ...})
	github.com/stackrox/rox/central/debug/service/service.go:559 +0xa5e
github.com/stackrox/rox/central/debug/service.(*serviceImpl).getDiagnosticDump(0xc0012f9000, {0x6be04f0, 0xc0069e7f20}, 0xc00716a100)
	github.com/stackrox/rox/central/debug/service/service.go:666 +0x170
net/http.HandlerFunc.ServeHTTP(0x1de66d4, {0x6be04f0, 0xc0069e7f20}, 0xc00600b410)
	net/http/server.go:2046 +0x2f
github.com/stackrox/rox/pkg/sac/observe.AuthzTraceHTTPInterceptor.func1.1({0x6be04f0, 0xc004674d20}, 0xc00716a100)
	github.com/stackrox/rox/pkg/sac/observe/interceptors.go:34 +0xdd
net/http.HandlerFunc.ServeHTTP(0x6c2c270, {0x6be04f0, 0xc004674d20}, 0xc0030a8780)
	net/http/server.go:2046 +0x2f
github.com/stackrox/rox/pkg/contextutil.HTTPInterceptor.func1.1({0x6be04f0, 0xc004674d20}, 0xc00716a000)
	github.com/stackrox/rox/pkg/contextutil/contextupdater.go:80 +0x250
net/http.HandlerFunc.ServeHTTP(0x6ba0de0, {0x6be04f0, 0xc004674d20}, 0x2)
	net/http/server.go:2046 +0x2f
github.com/stackrox/rox/pkg/grpc/routes.authorizerHandler.func1({0x6be04f0, 0xc004674d20}, 0xc00716a000)
	github.com/stackrox/rox/pkg/grpc/routes/routes.go:19 +0x1dd
net/http.HandlerFunc.ServeHTTP(0x6c2c270, {0x6be04f0, 0xc004674d20}, 0x7f0c599f4528)
	net/http/server.go:2046 +0x2f
github.com/stackrox/rox/pkg/contextutil.HTTPInterceptor.func1.1({0x6be04f0, 0xc004674d20}, 0xc0069ce400)
	github.com/stackrox/rox/pkg/contextutil/contextupdater.go:80 +0x250
net/http.HandlerFunc.ServeHTTP(0x6c2c1c8, {0x6be04f0, 0xc004674d20}, 0x98e3828)
	net/http/server.go:2046 +0x2f
github.com/stackrox/rox/pkg/grpc/requestinfo.(*Handler).HTTPIntercept.func1({0x6be04f0, 0xc004674d20}, 0xc0069ce100)
	github.com/stackrox/rox/pkg/grpc/requestinfo/requestinfo.go:288 +0x456
net/http.HandlerFunc.ServeHTTP(0x107f0caeaa3a68, {0x6be04f0, 0xc004674d20}, 0xc002980588)
	net/http/server.go:2046 +0x2f
github.com/stackrox/rox/pkg/grpc/metrics.(*httpMetricsImpl).WrapHandler.func1({0x6bf0300, 0xc0055ec0e0}, 0x5d22060)
	github.com/stackrox/rox/pkg/grpc/metrics/http_metrics_impl.go:65 +0x13a
net/http.HandlerFunc.ServeHTTP(0x16, {0x6bf0300, 0xc0055ec0e0}, 0x0)
	net/http/server.go:2046 +0x2f
net/http.(*ServeMux).ServeHTTP(0x0, {0x6bf0300, 0xc0055ec0e0}, 0xc0069ce100)
	net/http/server.go:2424 +0x149
golang.stackrox.io/grpc-http1/server.CreateDowngradingHandler.func1({0x6bf0300, 0xc0055ec0e0}, 0xc0069ce100)
	golang.stackrox.io/grpc-http1@v0.2.4/server/server.go:187 +0xe9
net/http.HandlerFunc.ServeHTTP(0x6bf0300, {0x6bf0300, 0xc0055ec0e0}, 0x4)
	net/http/server.go:2046 +0x2f
golang.org/x/net/http2/h2c.h2cHandler.ServeHTTP({{0x6ba0de0, 0xc0030e1f20}, 0xc0010e1600}, {0x6bf0300, 0xc0055ec0e0}, 0xc0069ce100)
	golang.org/x/net@v0.0.0-20220412020605-290c469a71a5/http2/h2c/h2c.go:104 +0x3e4
github.com/stackrox/rox/pkg/grpc.denyMisdirectedRequest.func1({0x6bf0300, 0xc0055ec0e0}, 0x10000c006f96d40)
	github.com/stackrox/rox/pkg/grpc/endpoints.go:119 +0x96
net/http.HandlerFunc.ServeHTTP(0x0, {0x6bf0300, 0xc0055ec0e0}, 0x1e4396e)
	net/http/server.go:2046 +0x2f
net/http.serverHandler.ServeHTTP({0xc004d3ae10}, {0x6bf0300, 0xc0055ec0e0}, 0xc0069ce100)
	net/http/server.go:2878 +0x43b
net/http.(*conn).serve(0xc006fc00a0, {0x6c2c270, 0xc001df81b0})
	net/http/server.go:1929 +0xb08
created by net/http.(*Server).Serve
	net/http/server.go:3033 +0x4e8
```